test/test_StateMachine.py

```diff
@@ -367,3 +367,34 @@ def test_reset(self):
         self.assertEqual(sm.state, 'bar')
         sm.reset()
         self.assertIsNone(sm.state)
+
+    def test_can(self):
+        sm = StateMachine({
+            'initial': 'init',
+            'transitions': {
+                ('init', 'foo'): lambda: True,
+                ('foo', 'bar'): lambda: True,
+                ('bar', 'foo'): lambda: True,
+                ('bar', 'init'): lambda: True
+            }
+        })
+
+        self.assertEqual(sm.state, None)
+        self.assertEqual(sm.can('init'), True)
+        self.assertEqual(sm.can('foo'), False)
+        self.assertEqual(sm.can('bar'), False)
+
+        sm.process(0)
+        self.assertEqual(sm.state, 'init')
+        self.assertEqual(sm.can('foo'), True)
+        self.assertEqual(sm.can('bar'), False)
+
+        sm.process(0)
+        self.assertEqual(sm.state, 'foo')
+        self.assertEqual(sm.can('bar'), True)
+        self.assertEqual(sm.can('init'), False)
+
+        sm.process(0)
+        self.assertEqual(sm.state, 'bar')
+        self.assertEqual(sm.can('foo'), True)
+        self.assertEqual(sm.can('init'), True)
```

Comment:
Arguably, these could use `self.assertTrue()` and `self.assertFalse()` respectively. Another alternative would be `self.assertIs()`.

I don't really have a preference one way or the other, just something to consider.

`self.assertIs()` would have the benefit of testing for identity with `True` and `False`; all the other asserts will succeed even if the value is only "truthy" or "falsy" (1 == True, 0 == False, etc).